Adjust vade to create presentation without proof

vineetpant opened the pull request:
## Description

This MR adds support for self issued presentation i.e Presentations without proof which contain unsigned or self issued credentials.

## Details

- add helper function `helper_create_selft_issued_presentations`
- add tests
- update bindings

vineetpant committed:
add helper_create_self_issued_presentation function

vineetpant committed:
update versions.md file

Tschuck commented on src/helpers/presentation.rs:
```diff
@@ -347,6 +363,50 @@ impl<'a> Presentation<'a> {
+        &mut self,
+        unsigned_credential_str: &str,
+    ) -> Result<String, PresentationError> {
+        if unsigned_credential_str.contains("proof") {
```
What happens, when the credential subject data contains a field named proof? :D

vineetpant replied:
makes sense, :) . changed the condition , now checking 
=> if credential is signed and finished then throw error otherwise proceed

Tschuck commented on src/helpers/presentation.rs:
```diff
@@ -637,6 +725,47 @@ mod tests_proof_request {
+        let self_presentation_result_str = self_presentation_result?;
+        assert!(
+            !self_presentation_result_str.contains("proof"),
+            "Self Issued Presentation cannot contian proof"
```
```suggestion
            "Self Issued Presentation can not contain proof"
```

vineetpant committed:
Update src/helpers/presentation.rs

Co-authored-by: Tschuck <7114819+Tschuck@users.noreply.github.com>

vineetpant committed:
update check for credential without proof

Tschuck committed:
Merge branch 'experimental/develop-next' into feature/adjust-vade-for-presentation-without-proof

wulfraem committed:
refactor check for existing proof

- now only checks for the `proof` field
- avoids confusing error messages when data structure is invalid

wulfraem committed:
refactor unnecessary `mut`
